Rsk.Samples.IdentityServer4.AdminUiIntegration/Program.cs

```diff
@@ -75,12 +75,19 @@ public static int Seed(ConfigurationDbContext context)
         {
             var missingClients = Clients.Where(x => !context.Clients.Any(y => y.ClientId == x.ClientId));
             context.Clients.AddRange(missingClients.Select(x => x.ToEntity()));
-            var missingIdentityResources = IdentityResources.Where(x => !context.IdentityResources.Any(y => y.Name == x.Name));
-            context.IdentityResources.AddRange(missingIdentityResources.Select(x => x.ToEntity()));
-            var missingApiResources = ApiResources.Where(x => !context.IdentityResources.Any(y => y.Name == x.Name));
+            var missingApiResources = ApiResources.Where(x => !context.ApiResources.Any(y => y.Name == x.Name));
             context.ApiResources.AddRange(missingApiResources.Select(x => x.ToEntity()));
-            var missingApiScopes = ApiScopes.Where(x => !context.IdentityResources.Any(y => y.Name == x.Name));
+            var missingApiScopes = ApiScopes.Where(x => !context.ApiScopes.Any(y => y.Name == x.Name));
             context.ApiScopes.AddRange(missingApiScopes.Select(x => x.ToEntity()));
+            var missingIdentityResources = IdentityResources.Where(x => !context.IdentityResources.Any(y => y.Name == x.Name));
+            List<Duende.IdentityServer.EntityFramework.Entities.IdentityResource> identityResourcesToAdd = new List<Duende.IdentityServer.EntityFramework.Entities.IdentityResource>();
+            foreach (var idr in missingIdentityResources)
+            {
+                var idrToEntity = idr.ToEntity();
+                if (idrToEntity.Name == IdentityServerConstants.StandardScopes.OpenId || idrToEntity.Name == IdentityServerConstants.StandardScopes.Profile) idrToEntity.NonEditable = true;
+                identityResourcesToAdd.Add(idrToEntity);
+            }
+            context.IdentityResources.AddRange(identityResourcesToAdd);
```

Comment:
**nit:** it's fine for you to add them as you go, you don't _need_ to batch it here. The database will only be hit on save changes.